packages/flutter_tools/test/commands.shard/permeable/build_bundle_test.dart

```diff
@@ -557,6 +557,238 @@ void main() {
+      'bundle',
+      '--no-pub',
+      '--dart-define-from-file=config1.json',
+    ]);
+  }, overrides: <Type, Generator>{
+    BuildSystem: () => TestBuildSystem.all(BuildResult(success: true),
+            (Target target, Environment environment) {
+          expect(
+            _decodeDartDefines(environment),
+            containsAllInOrder(const <String>[
+              'myVar=sfadsfv=432f',
+            ]),
+          );
+        }),
+    FileSystem: fsFactory,
+    ProcessManager: () => FakeProcessManager.any(),
+  });
+
+  testUsingContext('test --dart-define-from-file option env file', () async {
+    globals.fs
+        .file(globals.fs.path.join('lib', 'main.dart'))
+        .createSync(recursive: true);
+    globals.fs.file('pubspec.yaml').createSync();
+    globals.fs.file('.packages').createSync();
+    await globals.fs.file('.env').writeAsString('''
+        # comment
+        kInt=1
+        kDouble=1.1 # should be double
+
+        name=piotrfleury
+        title=this is title from config env file
+        empty=
+
+        doubleQuotes="double quotes 'value'#=" # double quotes
+        singleQuotes='single quotes "value"#=' # single quotes
+        backQuotes=`back quotes "value" '#=` # back quotes
+
+        hashString="some-#-hash-string-value"
+
+        # Play around with spaces around the equals sign.
+        spaceBeforeEqual =value
+        spaceAroundEqual = value
+        spaceAfterEqual= value
+
+      ''');
+    await globals.fs.file('.env2').writeAsString('''
+        # second comment
+
+        body=this is body from config env file
+      ''');
+    final CommandRunner<void> runner =
+        createTestCommandRunner(BuildBundleCommand(
+      logger: BufferLogger.test(),
+    ));
+
+    await runner.run(<String>[
+      'bundle',
+      '--no-pub',
+      '--dart-define-from-file=.env',
+      '--dart-define-from-file=.env2',
+    ]);
+  }, overrides: <Type, Generator>{
+    BuildSystem: () => TestBuildSystem.all(BuildResult(success: true),
+            (Target target, Environment environment) {
+          expect(
+            _decodeDartDefines(environment),
+            containsAllInOrder(const <String>[
+              'kInt=1',
+              'kDouble=1.1',
+              'name=piotrfleury',
+              'title=this is title from config env file',
+              'empty=',
+              "doubleQuotes=double quotes 'value'#=",
+              'singleQuotes=single quotes "value"#=',
+              'backQuotes=back quotes "value" \'#=',
+              'hashString=some-#-hash-string-value',
+              'spaceBeforeEqual=value',
+              'spaceAroundEqual=value',
+              'spaceAfterEqual=value',
+              'body=this is body from config env file'
+            ]),
+          );
+        }),
+    FileSystem: fsFactory,
+    ProcessManager: () => FakeProcessManager.any(),
+  });
+
+  testUsingContext('test --dart-define-from-file option env file with multi-line value', () async {
```

Comment:
Consider updating test title to say what we are testing for.

```suggestion
  testUsingContext('--dart-define-from-file option throws a ToolExit when .env file contains a multiline value', () async {
```